src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -359,18 +359,34 @@ internal Task<int> SendToAsync(ArraySegment<byte> buffer, SocketFlags socketFlag
             return tcs.Task;
         }
 
+        private static void ValidateBufferArguments(byte[] buffer, int offset, int size)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if ((uint)offset > (uint)buffer.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+            if ((uint)size > (uint)(buffer.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(size));
+            }
+        }
```

Comment:
Nit: shouldn't this go to `Socket.cs` instead?

Reply:
It could, but I put it here because a) the other buffer validation method is here right below it and b) this is about validating inputs when we're about to call task-based methods.  The split between these files is somewhat arbitrary as well; we could combine them later.

Reply:
> b) this is about validating inputs when we're about to call task-based methods

To me it seems like this particular validation method is only called by sync and APM overloads which both dwell in `Socket.cs`.

Reply:
Let's subsequently just combine the files.